lib/catalog/supervisor.ex

```diff
@@ -0,0 +1,17 @@
+defmodule Catalog.Supervisor do
+  use Supervisor
+
+  def start_link([]) do
+    Supervisor.start_link(__MODULE__, :ok)
+  end
+
+  @impl true
+  def init(:ok) do
+    children = [
+      Catalog,
+      Catalog.Health
```

Comment:
You want this to be the other way around. The Health server needs to run first because `Catalog` depends on being able to call the `loaded` function.

Reply:
This is very true.